massa-proto/proto/massa/api/v1/execution.proto

```diff
@@ -67,16 +70,226 @@ message ScExecutionEventContext {
 
 // ScExecutionEventStatus type enum
 enum ScExecutionEventStatus {
-  SC_EXECUTION_EVENT_STATUS_UNSPECIFIED = 0; // Defaut enum value
+  SC_EXECUTION_EVENT_STATUS_UNSPECIFIED = 0; // Default enum value
   SC_EXECUTION_EVENT_STATUS_FINAL = 1; // Final status
   SC_EXECUTION_EVENT_STATUS_READ_ONLY = 2; // Read only status
   SC_EXECUTION_EVENT_STATUS_FAILURE = 3; // Failure status
 }
 
 // ExecutionOutputStatus type enum
 enum ExecutionOutputStatus {
-  EXECUTION_OUTPUT_STATUS_UNSPECIFIED = 0; // Defaut enum value
+  EXECUTION_OUTPUT_STATUS_UNSPECIFIED = 0; // Default enum value
   EXECUTION_OUTPUT_STATUS_CANDIDATE = 1; // Candidate status
   EXECUTION_OUTPUT_STATUS_FINAL = 2; // Final status
 }
- 
\ No newline at end of file
+
+// StateChanges
+message StateChanges {
+  // Ledger changes
+  LedgerChanges ledger_changes = 1;
+  // Asynchronous pool changes
+  repeated AsyncPoolChangeEntry async_pool_changes = 2;
+  // Executed operations changes
+  ExecutedOpsChanges executed_ops_changes = 5;
```

Comment:
Why is it 5 there ?

Reply:
i think it should be set to 3

Reply:
> I would like to keep the fields order defined in the Rust struct

AFAIK at the binary level and unless explicitly stated otherwise, Rust does not guaranties the order of the fields. 
So I think you can relax this constraint :)

Reply:
It's just for readability purpose, I tried to keep gRPC messages as close as possible to their Rust structs equivalents

Reply:
Isn't having the same name enough?
Protobuf noob questions:
What if there is a refactor of the Rust struct and this field is replaced by another with a different meaning and we want to serialize it with protobuf too?

Reply:
@bilboquet It's not mandatory but nice to have. Like the alphabetic order of all rpc methods

Reply:
done